sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -105,11 +105,14 @@ class StreamExecution(
   var lastExecution: QueryExecution = null
 
   @volatile
-  var streamDeathCause: StreamingQueryException = null
+  private var streamDeathCause: StreamingQueryException = null
 
   /* Get the call site in the caller thread; will pass this into the micro batch thread */
   private val callSite = Utils.getCallSite()
 
+  private val streamMetrics = new StreamMetrics(uniqueSources.toSet, triggerClock,
+    "%s.StructuredStreamingMetrics.%s".format(sparkSession.sparkContext.appName, name))
```

Comment:
Changing the format. Tested on Ganglia and decided against putting appname inside since we already have the query name.